actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -25,22 +25,22 @@ def generate(name, options, path_parameters, parameterize = nil)
           next unless name || route.dispatcher?
 
           missing_keys = missing_keys(route, parameterized_parts)
-          next unless missing_keys.empty?
+          next if missing_keys && missing_keys.any?
```

Comment:
You're correct, I never knew. it looks like the big difference is this line: https://github.com/ruby/ruby/blob/trunk/array.c#L5524

I'm actually looking to see if we can optimize the case of `any?` a little for when the array is empty. It will still be slightly slowe though. I think it would make sense for this to be:

```ruby
if missing_keys && !missing_keys.empty?
```

Want to give me a PR and @ mention me?